README.md

```diff
@@ -11,46 +11,65 @@ This template includes:
 
-Add the following to your `.bashrc` file:
+Add the following to your `.bashrc` or `.zshrc` file:
 
-```
+```bash
+# Normal aliases for normal usage & interacting with online AWS
 alias node-run='docker run --rm -it --volume ~/.aws:/home/node/.aws --volume ~/.npm:/home/node/.npm --volume $PWD:/app aligent/serverless'
 alias serverless='node-run serverless'
-alias sls-deploy-local='docker-compose exec -u node -w /app offline /serverless/node_modules/serverless/bin/serverless.js deploy --verbose --aws-profile localstack --stage dev'
-alias sls-invoke='docker-compose exec -u node -w /app offline /serverless/node_modules/serverless/bin/serverless.js invoke --verbose --aws-profile localstack --stage dev --function'
-alias sls-invoke-stepf='docker-compose exec -u node -w /app offline /serverless/node_modules/serverless/bin/serverless.js invoke stepf --verbose --aws-profile localstack --stage dev --name'
+alias npm='node-run npm'
+alias aws='node-run aws'
+
+# Local aliases for interacting with localstack
+alias local-run='docker run --rm -it --volume ~/.aws:/home/node/.aws --volume ~/.npm:/home/node/.npm --volume $PWD:/app --network localstack-net aligent/serverless:latest'
+alias sls-local='local-run npm run serverless-local --'
+alias sls-local-deploy='sls-local deploy --verbose --aws-profile localstack --stage dev'
+alias sls-local-invoke='sls-local invoke --verbose --aws-profile localstack --stage dev'
+alias sls-local-invoke-stepf='sls-local invoke stepf --verbose --aws-profile localstack --stage dev'
+alias aws-local='local-run aws --endpoint-url=http://172.20.0.100:4566'
 ```
 
-You will then need to reload your bashrc file, either by running `. ~/.bashrc` or starting a new terminal session.
-
-Start with the template: `serverless create --template-url https://github.com/aligent/serverless-aws-nodejs-service-template --path my-project`
-
-Then install dependencies:
-
-`cd my-project`
-
-`node-run npm install`.
+#### Notes
+- You will then need to reload your bashrc file, either by running `. ~/.bashrc` (or `. ~/.zshrc`) or starting a new terminal session.
+- These aliases only works when you are inside the project folder.
+- For the local aliases to work, you need the localstack container running in background. Check Offline Usage for more details.
 
 ### Local NPM
 
 Install serverless globally see: https://www.serverless.com/framework/docs/getting-started/
 
-Start with the template: `serverless create --template-url https://github.com/aligent/serverless-aws-nodejs-service-template --path my-project`
+### Quick start
+Start with the template: 
+```bash
+serverless create --template-url https://github.com/aligent/serverless-aws-nodejs-service-template --path my-project
+```
 
 Then install dependencies:
+```bash
+cd my-project
+npm install
+```
 
-`cd my-project`
-
-`node-run npm install`.
+If you want to use the pre-configured VSCode debugging setup:
+```bash
+mkdir .vscode
+cp .vscode-configs/. .vscode
+```
```

Comment:
This feels like an annoying additional step. I understand it's because we want `.vscode` in gitignore.. is there a way to have it part of the template anyway? @TheOrangePuff any ideas? I'd like to minimise friction/setup after creating a service form the template as much as possible.

Reply:
We can have a bootstrap bash script that do this & update the aliases too. I didn't do that because I don't want to override `.vscode` other's folder in case they already have their own settings. With a bootstrap bash script we can:

1. Override `.vscode` folder.
2. Create/update `~/.aligent_aliases` file.
3. Update `.bashrc` to source `~/.aligent-serverless-aliases` so that each time we need to update or add more aliases, we don't have to modify `.bashrc` by ourselves.

Aliases that I'm having on my machine looks like this:
`kai.nguyen ~ ls -al
-rw-r--r--  1 kai.nguyen domain users   629 Aug 31 09:55 .aligent_aliases
-rw-r--r--  1 kai.nguyen domain users  4332 Aug 31 09:55 .zshrc

kai.nguyen ~ cat .aligent_aliases 
# Aliases for aligent/serverless-aws-nodejs-service-template
alias sls-run='docker run --rm -it --volume ~/.aws:/home/node/.aws --volume ~/.azure:/home/node/.azure --volume ~/.npm:/home/node/.npm --volume "$PWD:/app" aligent/serverless:latest'
alias serverless='sls-run serverless'
alias npm='sls-run npm'
#
# Aliases for running against localstack
alias local-run='docker run --rm -it --volume ~/.aws:/home/node/.aws --volume ~/.azure:/home/node/.azure --volume ~/.npm:/home/node/.npm --volume "$PWD:/app" --network localstack-net aligent/serverless:latest'
alias sls-local='local-run serverless'
alias aws-local='local-run aws'

kai.nguyen ~ cat .zshrc
# Aligent's aliases
if [ -f ~/.aligent_aliases ]; then
    . ~/.aligent_aliases
fi`

Reply:
Why can't we just commit `.vscode` to this repo? Then all these configs would be loaded automatically. Because it's only project level config not system wide config I think committing it is fine.

I don't love the idea of a script that modifies `.bashrc`... that can get messy.

Aliases are one of those things that are pretty hard to manage at a company wide level. I have something similar to what you suggested in my `zshrc` that links to Aligent specific aliases. Maybe this is something sysadmin can put on the default image? I know that doesn't help in the immediate but for future employees it could help. 
But then updating the Aligent aliases is a whole different thing...

Reply:
I agree that modifying `.bashrc` can be messy too. That's why I don't like the idea of manually modify it as not everyone understand how this `.bashrc` thing works. It's also time consuming to fix if an error arise because we update our template but the aliases are not. Maybe we can have a script that update the aliases in `.aligent_aliases` (or should it be `.aligent_serverless_aliases` for serverless aliases only). User or sysadmin can modify their `.bashrc` manually (it's just an if statement that does not change frequently).

Reply:
My experience with git is that if we commit `.vscode` to this repo and _then_ commit the .gitignore change the `.vscode` folder will stick, right? But then I'm not sure you can commit any changes which isn't great.

Reply:
Also @kai-nguyen-aligent I think it needs to be `cp -r .vscode-configs/. .vscode`

Reply:
> My experience with git is that if we commit `.vscode` to this repo and _then_ commit the .gitignore change the `.vscode` folder will stick, right? But then I'm not sure you can commit any changes which isn't great.
Yeah, we will not be able to commit changes unless we remove it from `.gitignore` file.


> Also @kai-nguyen-aligent I think it needs to be `cp -r .vscode-configs/. .vscode`
I added that flag but it will not make any difference because we don't have nested folder inside `.vscode`